### PR TITLE
run_iodemo cleanups

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by yosefe** [comment]
@mike-dubman can you pls take a look

File: test/apps/iodemo/run_io_demo.sh [hunk 1]
```diff
@@ -222,9 +222,12 @@ parse_args()
 
 check_params()
 {
-	if [ ! -x "${iodemo_exe}" ]
+	if [ -z "${iodemo_exe}" ]
```
Comment (mike-dubman) [inline test/apps/iodemo/run_io_demo.sh:225]: ```-z``` will check if var has value, while we want to check that variable is executable (iodemo_exe)

File: test/apps/iodemo/run_io_demo.sh [hunk 2]
```diff
@@ -222,9 +222,12 @@ parse_args()
...
 	then
 		error "missing executable from command line"
+	elif [ ! -x "${iodemo_exe}" ]
```
Comment (mike-dubman) [inline test/apps/iodemo/run_io_demo.sh:228]: IMO -x implies -z as well and -z check is redundant

Comment (yosefe) [reply test/apps/iodemo/run_io_demo.sh:228]: the error message was not accurate

File: test/apps/iodemo/run_io_demo.sh [hunk 3]
```diff
@@ -414,238 +417,238 @@ make_scripts()
...
+			    for pid in \$(list_pids)
+			    do
+			        grep -qP "IODEMO_ROLE=\$1\x00" /proc/\${pid}/environ \\
+			            && echo \${pid}
+			    done
+			}
+
+			kill_iodemo() {
+			    pids="\$(list_pids)"
+			    [ -n "\${pids}" ] && kill -9 \${pids}
```
Comment (mike-dubman) [inline test/apps/iodemo/run_io_demo.sh:444]: btw, is there signal that iodemo_exe will respond with graceful exit?

Comment (yosefe) [reply test/apps/iodemo/run_io_demo.sh:444]: there is no such signal, we could add it and use kill -9 after some timeout (to ensure cleanup)

File: test/apps/iodemo/run_io_demo.sh [hunk 4]
```diff
@@ -414,238 +417,238 @@ make_scripts()
...
+			    echo "Usage: ${command_file} [options] "
+			    echo
+			    echo "Where options are:"
+			    echo "  -h                 Show this help message"
+			    echo "  -list-tags         List available tags and exit"
+			    echo "  -start <tag>       Start iodemo for given tag"
+			    echo "  -stop <tag>        Stop iodemo for given tag"
+			    echo "  -status <tag>      Show status of iodemo for given tag"
+			    echo
+			    echo "If no optoins are given, run all commands and wait for completion"
```
Comment (mike-dubman) [inline test/apps/iodemo/run_io_demo.sh]: s/optoins/options

File: test/apps/iodemo/run_io_demo.sh [hunk 5]
```diff
@@ -414,238 +417,238 @@ make_scripts()
...
+			    echo "Test finished"
+			}
+
 			EOF
-		cat >>${command_file} <<-'EOF'
-		parse_args "$@"
-		kill_iodemo
-		main
+
+		cat >>${command_file} <<-EOF
```
Comment (mike-dubman) [inline test/apps/iodemo/run_io_demo.sh:614]: you can use ```<<-'EOF'``` and no need to wrap ```\$```

Comment (yosefe) [reply test/apps/iodemo/run_io_demo.sh:614]: in some place i want to replace vars from external script, so this keeps everything consistent

**Review by mike-dubman — APPROVED** [review]

**Pull request merged**